src/evict/evict_lru.c

```diff
@@ -2254,6 +2258,10 @@ __evict_page(WT_SESSION_IMPL *session, bool is_server)
         }
     }
 
+    /* Set a flag to indicate that either eviction server or worker thread is evicting the page. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL))
```

Comment:
Why we can't use WT_SESSION_INTERNAL but introduce WT_REC_EVICTION_THREAD? Logically, they seem to be the same.

Reply:
We can only be in `__evict_page()` either in context of application threads or eviction threads. The presence of flag `WT_SESSION_INTERNAL` further narrows the scenario down to eviction threads.